include/ocpp/common/message_queue.hpp

```diff
@@ -340,46 +381,49 @@ template <typename M> class MessageQueue {
                 }
 
                 // prioritize the message with the oldest timestamp
-                auto now = DateTime();
```

Comment:
Maybe the idea of a single timestamp was not that wrong, you can also pass it inside the lambda.